### PR TITLE
Add types from DefinitelyTyped

### PR DESCRIPTION
**Pull request by thegecko**
This PR adds in the types as defined on [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/noble)

This allows the library to be easily used in TypeScript projects

### CODE REVIEW

**Comment by rzr** [comment]
reviewers welcome

**Comment by thegecko** [comment]
bump

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    type: string;
+    includedServiceUuids: string[];
+    characteristics: Characteristic[];
+
+    discoverIncludedServices(serviceUUIDs: string[], callback?: (error: string, includedServiceUuids: string[]) => void): void;
+    discoverCharacteristics(characteristicUUIDs: string[], callback?: (error: string, characteristics: Characteristic[]) => void): void;
+    toString(): string;
+
+    on(event: "includedServicesDiscover", listener: (includedServiceUuids: string[]) => void): this;
+	on(event: "characteristicsDiscover", listener: (characteristics: Characteristic[]) => void): this;
```
Comment (rzr) [inline index.d.ts]: is indenting correct ? if not please add an other patch to check formating using eslint

Comment (thegecko) [reply index.d.ts]: It was copied directly from DefinitelyTyped ;), have updated formatting

**Comment by thegecko** [comment]
bump

**Comment by rzr** [comment]
Is anyone is against merging this ?

@abandonware/reviewers 

Meanwhile please create a bug on upstream tracker
then cross link to this 
https://github.com/abandonware/noble/pull/31
in commit message
 
More hints at:
https://www.slideshare.net/rzrfreefr/updownstreamflows20190411rzr#

**Comment by mrstegeman** [comment]
No issues here.

**Pull request merged**